core/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -407,11 +407,10 @@ private ThreadContextStruct putHeaders(Map<String, String> headers) {
             if (headers.isEmpty()) {
                 return this;
             } else {
-                final Map<String, String> newHeaders = new HashMap<>();
+                final Map<String, String> newHeaders = new HashMap<>(this.requestHeaders);
```

Comment:
I know what you mean now. You don't want to thrown an exception.
I found it throw an exception in the putRequest method, So I did the same.
I've also found that in the stashAndMergeHeaders method,  if a key exists in context.requestHeaders and headers,  the value from requestHeaders would have "won".
I'm not sure now.  I hope you can give me a conclusion.  if a key exists in requestHeaders and headers, whether requestHeaders to overwrite headers or headers to overwrite  requestHeaders.
Thank you.